ui/raidboss/browser_tts_engine.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+const TTSEngineType = {
+  SpeechSynthesis: 0,
+  GoogleTTS: 1,
+};
+
+class TTSItem {
+  play() {}
+}
+
+class SpeechTTSItem extends TTSItem {
+  constructor(text) {
+    super();
+    this.text = text;
+    this.item = new SpeechSynthesisUtterance(text);
+  }
+
+  play() {
+    window.speechSynthesis.speak(this.item);
+  }
+}
+
+class GoogleTTSItem extends TTSItem {
+  constructor(text) {
+    super();
+    this.text = text;
+    let iframe = document.createElement('iframe');
+    // remove sandbox so we can modify contents/call play on audio element later
+    iframe.removeAttribute('sandbox');
+    iframe.style.display = 'none';
+    document.body.appendChild(iframe);
+    let encText = encodeURIComponent(text);
+    iframe.contentDocument.body.innerHTML = '<audio src="https://translate.google.com/translate_tts?ie=UTF-8&client=tw-ob&tl=en&q=' + encText + '" id="TTS">';
+    this.item = iframe.contentDocument.body.firstElementChild;
+  }
+
+  play() {
+    this.item.play();
+  }
+}
+
+class BrowserTTSEngine {
+  constructor() {
+    // figure out what TTS engine type we need
+    if (window.speechSynthesis !== undefined) {
+      let me = this;
```

Comment:
If you change the `function() {` on the next line to `() => {`, then it won't rebind this and you can just say `this.engineType` inside of the function without the extra variable.

Reply:
Fixed in latest changes, thanks.